src/core/server/index.ts

```diff
@@ -221,14 +222,30 @@ class Server {
     await this.i18n.load();
 
     // Setup MongoDB.
-    this.mongo = await createMongoDB(config);
+    const liveURI = config.get("mongodb");
+    const live = await createMongoDB(liveURI);
+
+    // If we have an archive URI, use it, otherwise, default
+    // to using the live database
+    let archive: Db | null = null;
+    const archiveURI = config.get("mongodb_archive");
+    if (
+      archiveURI === config.default("mongodb_archive") &&
+      liveURI !== config.default("mongodb")
+    ) {
+      archive = live;
+    } else {
+      archive = await createMongoDB(archiveURI);
+    }
+
+    this.mongo = { live, archive };
```

Comment:
Move the creation logic inside the context, so you'll have a `createMongoContext(config)`